crates/chain/src/keychain/txout_index.rs

```diff
@@ -444,45 +458,42 @@ impl<K: Clone + Ord + Debug> KeychainTxOutIndex<K> {
         keychain: K,
         descriptor: Descriptor<DescriptorPublicKey>,
     ) -> super::ChangeSet<K> {
-        let descriptor_id = descriptor.descriptor_id();
-        // First, we fill the keychain -> (desc_id, descriptor) map
-        let old_descriptor_opt = self
+        let mut changeset = super::ChangeSet::<K>::default();
+        let desc_id = descriptor.descriptor_id();
+
+        let old_desc = self
             .keychains_to_descriptors
-            .insert(keychain.clone(), (descriptor_id, descriptor.clone()));
-
-        // Then, we fill the descriptor_id -> (keychain, descriptor) map
-        let old_keychain_opt = self
-            .descriptor_ids_to_keychain
-            .insert(descriptor_id, (keychain.clone(), descriptor.clone()));
-
-        // If `keychain` already had a `descriptor` associated, different from the `descriptor`
-        // passed in, we remove it from the descriptor -> keychain map
-        if let Some((old_desc_id, _)) = old_descriptor_opt {
-            if old_desc_id != descriptor_id {
-                self.descriptor_ids_to_keychain.remove(&old_desc_id);
+            .insert(keychain.clone(), (desc_id, descriptor.clone()));
```

Comment:
The claim of the commit message was that the structure is now fully monotone. But it's not, this function blindly changes the descriptor associated with a keychain. The keychian <-> descriptor relationship will depend on the order of the changessets/calls to this.